test/test-tcp-try-write.c

```diff
@@ -43,12 +43,46 @@ static void close_cb(uv_handle_t* handle) {
 }
 
 
+static void bad_write_cb(uv_write_t *req, int status) {
+  ASSERT(0); // should be unreachable
+}
+
+
+static void bad_shutdown_cb(uv_shutdown_t *req, int status) {
+  ASSERT(0); // should be unreachable
+}
+
+
+static void bad_alloc_cb(uv_handle_t* handle, size_t size, uv_buf_t* buf) {
+  ASSERT(0); // should be unreachable
+}
+
+
+static void bad_read_cb(uv_stream_t* stream, ssize_t nread, const uv_buf_t* buf) {
+  ASSERT(0); // should be unreachable
+}
+
+
 static void connect_cb(uv_connect_t* req, int status) {
+  static uv_write_t write_req;
+  static uv_shutdown_t shutdown_req;
```

Comment:
These don't need to be `static`, right?

Reply:
no, but if the asserts did somehow manage to fail, I figured it might be better for libuv not to be pointing to some stack memory